### PR TITLE
feat(datepicker): Making it possible to disable certain days in the DatePicker

### PR DESCRIPTION
**Pull request by sanguine-my-brother**
I've added the ability to disable certain days in the calendar. The code is based on the old module where it had an input named daysDisabled. This input was a number array which allowed you to specifiy which days of the week should be disabled. 

I wasn't sure where I should place an unit test, but feel free to notify me about where this should be done.

# PR Checklist
Before creating new PR, please take a look at checklist below to make sure that you've done everything that needs to be done before we can merge it.

 - [X] read and followed the [CONTRIBUTING.md](https://github.com/valor-software/ngx-bootstrap/blob/development/CONTRIBUTING.md) guide.
 - [X] built and tested the changes locally.
 - [ ] added/updated tests.
 - [x] added/updated API documentation.
 - [X] added/updated demos.

### CODE REVIEW

**Comment by mrVers** [comment]
Looking forward for this feature. Hope it gets into the next release 👍

**Comment by Kirschbaumke** [comment]
This would be a great feature, especially in our region where the Monday is the first day of the week.

**Comment by hardikjain90** [comment]
When this pull request gets accepted?

**Comment by loonix** [comment]
this is a very good feature, would be great if this could be merged.

File: src/chronos/utils/date-compare.ts [hunk 1]
```diff
@@ -33,6 +33,11 @@ export function isBefore(
   return endOf(date1, units).valueOf() < date2.valueOf();
 }
 
+export function isDisabledDay(date: Date, daysDisabled: number[]):boolean{
+  if(daysDisabled === undefined) return false;
+  return (daysDisabled.filter(d => d == date.getDay()).length > 0);
```
Comment (valorkin) [inline src/chronos/utils/date-compare.ts]: Replace with `.any()` we don't actually need an array of results, we just need to know that there are at least one match.
Plus if disabled days array is empty we can `return false;` too

**Review by valorkin — CHANGES_REQUESTED** [review]
With a small change should be good to go,
Tests is missing tho.
Could be a good base for disabled weekends PR

**Comment by sanguine-my-brother** [comment]
@valorkin will add these changes in the evening today.

**Comment by valorkin** [comment]
@eternia16 I have asked @domainv to apply my comments, so I could release this PR today after manual testing.
Please write tests ;)
It will be a separate PR most probanly

File: src/chronos/utils/date-compare.ts [hunk 2]
```diff
@@ -33,6 +33,20 @@ export function isBefore(
   return endOf(date1, units).valueOf() < date2.valueOf();
 }
 
+export function isDisabledDay(date: Date, daysDisabled: number[]): boolean {
+  if (daysDisabled === undefined || !daysDisabled.length) {
+    return false;
+  }
+
+  return (
+    daysDisabled.some(
```
Comment (ludmilanesvitiy) [inline src/chronos/utils/date-compare.ts]:
```suggestion
    daysDisabled.some((day: number) => day === date.getDay());
```
and remove what's benith
wrote by @valorkin

**Review by ludmilanesvitiy — CHANGES_REQUESTED** [review]

File: src/chronos/utils/date-compare.ts (hunk 2 above)
Comment (Domainv) [reply src/chronos/utils/date-compare.ts]: @ludmilanesvitiy 
Dima was not talking about removing, but about replacing filter on some method.

**Comment by ludmilanesvitiy** [comment]
Datepicker demo with Days Disabled tested in latest Chrome, FF. Exist issue with hover on disabled days, but it'll be fixed in other PR's. RFM

**Comment by ludmilanesvitiy** [comment]
Datepicker demo with Days Disabled tested in latest Chrome, FF. Exist issue with hover on disabled days, but it'll be fixed in other PR's. RFM

**Review by valorkin — APPROVED** [review]

**Comment by daniloff200** [comment]
closes #4531

**Pull request merged**